subsys/bluetooth/host/l2cap.c

```diff
@@ -3119,57 +3114,43 @@ int bt_l2cap_chan_disconnect(struct bt_l2cap_chan *chan)
 static int bt_l2cap_dyn_chan_send(struct bt_l2cap_le_chan *le_chan, struct net_buf *buf)
 {
 	struct l2cap_tx_meta_data *data;
-	void *old_user_data = l2cap_tx_meta_data(buf);
-	uint16_t sdu_len;
-	int err;
 
-	sdu_len = net_buf_frags_len(buf);
+	LOG_DBG("chan %p buf %p", le_chan, buf);
 
-	if (sdu_len > le_chan->tx.mtu) {
+	if (net_buf_frags_len(buf) > le_chan->tx.mtu) {
+		LOG_ERR("attempt to send %u bytes on %u MTU chan",
+			net_buf_frags_len(buf), le_chan->tx.mtu);
 		return -EMSGSIZE;
 	}
 
+	if (net_buf_headroom(buf) < BT_L2CAP_SDU_CHAN_SEND_RESERVE) {
+		/* Call `net_buf_reserve(buf, BT_L2CAP_SDU_CHAN_SEND_RESERVE)`
+		 * before passing buffer to L2CAP.
```

Comment:
Quality-of-life-thing: Let's not lead people into the mess-up-my-data footgun. This will save the developer one cycle of type-compile-run-test time.

```suggestion
		 * when allocating buffers intended for L2CAP.
```